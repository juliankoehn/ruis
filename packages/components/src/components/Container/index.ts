export { default as Container } from './src/Container';
