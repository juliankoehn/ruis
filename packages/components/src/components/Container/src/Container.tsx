import React from 'react';
import { css } from '@emotion/core';
import tokens from '@ruids/tokens';

const ContainerStyles = css({
  width: '100%',
  marginRight: 'auto',
  marginLeft: 'auto',
  paddingLeft: tokens.spacing1,
  paddingRight: tokens.spacing1,
});

const fluidStyles = css`
  @media (min-width: ${tokens.screenSm}) {
    max-width: ${tokens.screenSm};
  }
  @media (min-width: ${tokens.screenMd}) {
    max-width: ${tokens.screenMd};
  }
  @media (min-width: ${tokens.screenLg}) {
    max-width: ${tokens.screenLg};
  }
  @media (min-width: ${tokens.screenXl}) {
    max-width: ${tokens.screenXl};
  }
`;

const Container = React.forwardRef(
  (
    props: { fluid?: boolean; children: React.ReactNode },
    ref: React.Ref<HTMLDivElement>,
  ) => {
    const { fluid, ...rest } = props;
    const styles = [ContainerStyles];
    if (!fluid) {
      styles.push(fluidStyles);
    }
    return <div ref={ref} css={styles} {...rest} />;
  },
);

export default Container;
