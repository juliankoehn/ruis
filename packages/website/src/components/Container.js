import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@emotion/core';
import tokens from '@ruids/tokens';
import { heightOfHeader } from './Navigation/Navigation';
import { MDXProvider } from '@mdx-js/react';

const styles = {
    container: css`
    width: 100%;
    display: flex;
    flex-direction: column;
    height: calc(100vh - ${heightOfHeader}px);
    overflow-y: auto;
  `,

    main: css`
    flex: 1 1 0;
  `,

    inner: css`
    width: 960px;
    margin: 0 auto;
    padding: ${tokens.spacing12} ${tokens.spacing6} ${tokens.spacing6};
  `,

    innerHomePage: css`
    width: auto;
    text-align: center;
  `,
};

const Container = props => {
    const { frontmatter, children } = props;
    const isHomePage = frontmatter && frontmatter.type === 'home';


    return (
        <div css={styles.container}>
            <div css={styles.main}>
                <div css={isHomePage ? styles.innerHomePage : styles.inner}>
                    {children}
                </div>
            </div>
            
        </div>
    )
}

Container.propTypes = {
    children: PropTypes.node.isRequired,
};

export default Container