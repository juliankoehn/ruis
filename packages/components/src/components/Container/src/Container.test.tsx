import React from 'react';
import { render } from '@testing-library/react';
import Container from './Container';

describe('Container', () => {
  it('Should render a container', () => {
    let title = 'Test';
    const { container } = render(<Container>{title}</Container>);

    expect(container).toMatchSnapshot();
  });
});
