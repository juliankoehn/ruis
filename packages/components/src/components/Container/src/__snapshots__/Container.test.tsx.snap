// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Should render a container 1`] = `
<div>
  <div
    class="css-5x7jti-ContainerStyles-fluidStyles"
  >
    Test
  </div>
</div>
`;
